Spanner/tests/System/AdminTest.php

```diff
@@ -127,10 +129,74 @@ public function testDatabase()
         $this->assertEquals($db->ddl()[0], $stmt);
     }
 
-    public function testPgDatabase()
+    public function testCustomerManagedInstanceConfigurations()
     {
         $this->skipEmulatorTests();
         
+        $client = self::$client;
+
+        // Custom instance configuration IDs must start with 'custom' and may not contain any underscores.
+        $customConfigId = uniqid('custom-' . str_replace('_', '-', self::TESTING_PREFIX));
+
+        // Find the first instance configuration that has at least one optional replica. This indicates that it is a
+        // Google Managed multi-region config that can be used as the base config for a customer managed configuration.
+        $configurations = iterator_to_array($client->instanceConfigurations());
+        $baseConfig = array_filter($configurations, function ($configuration) {
+            return !empty($configuration->info()['optionalReplicas']);
+        });
+        if (empty($baseConfig)) {
+            $this->fail('No suitable base configuration found to create a custom instance config');
+        }
+        $baseConfig = $baseConfig[0];
```

Comment:
This might not work always as `array_filter` retains the keys, so it's not necessary that 0th offset will exist.
Once could use `array_shift` but if we use a for loop like suggested above, we won't need it.

Reply:
Done